android/titanium/src/java/org/appcelerator/titanium/TiApplication.java

```diff
@@ -831,6 +835,35 @@ private void stopExternalStorageMonitor()
 		unregisterReceiver(externalStorageReceiver);
 	}
 
+	private void startLockScreenMonitor()
+	{
+		lockScreenReceiver = new BroadcastReceiver() {
+			@Override
+			public void onReceive(Context context, Intent intent)
+			{
+				unlockedScreen = true;
+			}
+		};
+
+		IntentFilter filter = new IntentFilter();
+		filter.addAction(Intent.ACTION_SCREEN_ON);
+		registerReceiver(lockScreenReceiver, filter);
+	}
```

Comment:
One thing that bothers me is if `hasUnlockedScreen()` was called twice, it'll get reset to `false` on the second call. So, I'm thinking we should listen to both `ACTION_SCREEN_ON` and `ACTION_SCREEN_OFF`. Then in the `onReceive()`, you can set the `unlockedScreen` member variable based on which intent action was received like this...
```java
public void onReceive(Context context, Intent intent)
{
	switch (intent.getAction()) {
		case Intent.ACTION_SCREEN_ON:
			unlockedScreen = true;
			break;
		case Intent.ACTION_SCREEN_OFF:
			unlockedScreen = false;
			break;
	}
}
```

Reply:
And maybe we should add the following the if-check to the top of this method in case`startLockScreenMonitor()` was called twice. This will make it bullet proof and prevent a hanging listener that'll continue to receive intents.
```java
if (lockScreenReceiver != null) {
	return;
}
```

The rest looks great!